pkg/clusterrouter/proxy/proxy_server.go

```diff
@@ -233,12 +234,20 @@ func (r *RemoteService) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 			er.Error(rw, req, validation.Unauthorized)
 			return
 		}
-		token, err := r.getImpersonatorAccountToken(userInfo)
-		if err != nil && !strings.Contains(err.Error(), dialer2.ErrAgentDisconnected.Error()) {
-			er.Error(rw, req, fmt.Errorf("unable to create impersonator account: %w", err))
-			return
+
+		if !authcontext.IsSAAuthenticated(req.Context()) {
+			// If the request is not authenticated as a service account,
+			// we need to use an impersonation token.
+			// This is because the impersonator service account does exist on the downstream cluster, and
+			// it has sufficient permissions to perform the TokenReview.
+			token, err := r.getImpersonatorAccountToken(userInfo)
+			if err != nil && !strings.Contains(err.Error(), dialer2.ErrAgentDisconnected.Error()) {
```

Comment:
I appreciate that this is not a change in this PR, and...there are no tests for this code...

But could this use `errors.Is()` instead of the string comparisons?

Reply:
Using strings.Contains does seems like an odd choice here.